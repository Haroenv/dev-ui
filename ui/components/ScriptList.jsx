import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import ListItem from './ListItem';
const Container = styled.div``;

const ScriptList = ({ scripts = [], onScriptClick = () => {} }) => (
  <Container>
    {scripts.map(({ name, command }) => (
      <ListItem
        title={name}
        subtitle={command}
        key={name}
        buttons={<button onClick={() => onScriptClick(name)}>run</button>}
      />
    ))}
  </Container>
);

ScriptList.propTypes = {
  scripts: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      command: PropTypes.string.isRequired,
    }),
  ),
  onScriptClick: PropTypes.func,
};

export default ScriptList;
