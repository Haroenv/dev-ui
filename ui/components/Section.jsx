import styled from 'styled-components';

const Section = styled.section`
  margin: 0.5em;
  padding: 0.5em;
  border-radius: 0.2em;
  background: #eee;
  &:hover {
    background: #ddd;
    cursor: pointer;
  }
`;

export default Section;
