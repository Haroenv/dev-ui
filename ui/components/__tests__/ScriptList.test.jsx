import React from 'react';
import {shallow} from 'enzyme';

import ScriptList from '../ScriptList';
import Section from '../Section';

describe('<ScriptList />', () => {
  test('empty', () => {
    const wrapper = shallow(<ScriptList scripts={[]} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('with scripts', () => {
    const scripts = [
      {name: 'test', command: 'jest'},
      {name: 'lint', command: 'eslint .'},
    ];
    const wrapper = shallow(<ScriptList scripts={scripts} />);

    expect(wrapper.find(Section)).toHaveLength(2);
    expect(wrapper.find(Section).first()).toHaveStyleRule('margin', '0.5em');

    expect(wrapper).toMatchSnapshot();
  });
});
