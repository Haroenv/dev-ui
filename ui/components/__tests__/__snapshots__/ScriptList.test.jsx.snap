// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScriptList /> empty 1`] = `<styled.div />`;

exports[`<ScriptList /> with scripts 1`] = `
<styled.div>
  <styled.section
    key="test"
  >
    <styled.h2>
      test
    </styled.h2>
    <styled.div>
      jest
    </styled.div>
  </styled.section>
  <styled.section
    key="lint"
  >
    <styled.h2>
      lint
    </styled.h2>
    <styled.div>
      eslint .
    </styled.div>
  </styled.section>
</styled.div>
`;
